### DISCUSSION TITLE
Disable aggregate for one task in one subproject?

### DISCUSSION THREAD

**Discussion by swsnr** [sbt/sbt#6885] [Q&A] [answered]
Can I exclude one task in one sub-project of a multi-project SBT build from aggregate?  In other words, have `sbt compile` and `sbt test` aggregate over all sub-projects as given in the root `.aggregate()`, but have `sbt IntegrationTest/test` aggregate over all projects _expect_ one specific project?

Background is that I have a large multi-project build, where one or two sub-projects have special integration tests which need extra care and should run separately.  So far I've explicitly listed all sub-projects when running integration tests (e.g. `sbt projectA/IntegrationTest/test projectB/IntegrationTest/test`) but this obviously gets old quite soon, so I'd like to use aggregates, expect for a few select sub-projects, which should still be included in a regular `compile` or `test` though.

Is this possible in SBT? Or is there some kind of easy work-around? (I considered making my own test configuration for those "bad" sub-projects, but I'm not sure I find this "easy" :innocent:).

**Comment by eed3si9n** [comment, marked as answer]
Take a look at https://github.com/sbt/sbt/blob/f12a199a1aa58496ade139adcf8a7a8eba94b968/build.sbt#L1294-L1302

**Reply by swsnr** [reply]
I am afraid my SBT foo is not good enough to make sense of this 😬

Would you mind to explain this a bit? 😊

**Reply by eed3si9n** [reply]
That's an example of creating a dummy subproject, in this case named `upperModules` that aggregates all projects except for a few (lowerUtilProjects + bundledLauncherProj). You can then use that as a way of broadcasting commands like `upperModules/IntegrationTest/test` etc.

**Reply by swsnr** [reply]
Oh, now I see.  The idea of aggregating into a separate project didn't even occur to me :bulb:  :pray:
